scanners/ionic/ionic.go

```diff
@@ -75,56 +75,28 @@ func (scanner *Scanner) DetectPlatform(searchDir string) (bool, error) {
 		return false, fmt.Errorf("failed to search for files in (%s), error: %s", searchDir, err)
 	}
 
-	// Search for config.xml file
-	log.TInfof("Searching for config.xml file")
-
-	configXMLPth, err := cordova.FilterRootConfigXMLFile(fileList)
-	if err != nil {
-		return false, fmt.Errorf("failed to search for config.xml file, error: %s", err)
-	}
-
-	log.TPrintf("config.xml: %s", configXMLPth)
-
-	if configXMLPth == "" {
-		log.TPrintf("platform not detected")
-		return false, nil
-	}
-
-	widget, err := cordova.ParseConfigXML(configXMLPth)
-	if err != nil {
-		log.TPrintf("can not parse config.xml as a Cordova widget, error: %s", err)
-		log.TPrintf("platform not detected")
-		return false, nil
-	}
-
-	// ensure it is a cordova widget
-	if !strings.Contains(widget.XMLNSCDV, "cordova.apache.org") {
-		log.TPrintf("config.xml propert: xmlns:cdv does not contain cordova.apache.org")
-		log.TPrintf("platform not detected")
-		return false, nil
-	}
-
-	// ensure it is an ionic project
-	projectBaseDir := filepath.Dir(configXMLPth)
-
-	ionicProjectExist, err := pathutil.IsPathExists(filepath.Join(projectBaseDir, "ionic.project"))
+	// Ensure it is an ionic project
+	ionicConfigPath, err := FilterRootFile(fileList, "ionic.config.json")
 	if err != nil {
 		return false, fmt.Errorf("failed to check if project is an ionic project, error: %s", err)
 	}
 
-	ionicConfigExist, err := pathutil.IsPathExists(filepath.Join(projectBaseDir, "ionic.config.json"))
-	if err != nil {
-		return false, fmt.Errorf("failed to check if project is an ionic project, error: %s", err)
+	// Check the existence of the old ionic.project file
+	if ionicConfigPath == "" {
+		ionicConfigPath, err = FilterRootFile(fileList, "ionic.project")
+		if err != nil {
+			return false, fmt.Errorf("failed to check if project is an ionic project, error: %s", err)
+		}
 	}
 
-	if !ionicProjectExist && !ionicConfigExist {
-		log.Printf("no ionic.project file nor ionic.config.json found, seems to be a cordova project")
+	if ionicConfigPath == "" {
```

Comment:
What is the ionic.project's path (relative), is it is in the root directory?

Reply:
It's relative, but includes the file name as well, so it should be fine I guess, shouldn't it?

Reply:
It should be, yes. Usually it is better to use absolute path, where possible.

Reply:
Ah ok, thanks! TBH I just copied/modified the existing code here, didn't think about whether it can be a problem, but thanks for highlighting!